File: 완주하지못한선수.py

def solution(participant, completion):
    answer = ''
    part = dict() # 파이썬에서는 Dictionary라는 자료구조를 통해 해시 제공 => Dictionary는 dict클래스에 구현되어 있음
    for i in participant:
        if i in part:
            part[i] +=1
        else:
            part[i] = 1
    for i in completion:
        part[i] -= 1
    for i in part:
        if part[i] != 0:
            answer = i 
    return answer
  

File: 완주하지못한선수.java
import java.util.HashMap;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> hash= new HashMap<>();
        for(int i=0;i<participant.length;i++){
            if(hash.get(participant[i]) == null) hash.put(participant[i], 1);
            else hash.put(participant[i], hash.get(participant[i])+1);
         }
        for(int i=0;i<completion.length;i++){
            hash.put(completion[i], hash.get(completion[i])-1);
         }
        for (int i=0; i<participant.length; i++){
            if (hash.get(participant[i]) > 0) {
                answer = participant[i];
                break;
            }
        }
        return answer;
    }
}
